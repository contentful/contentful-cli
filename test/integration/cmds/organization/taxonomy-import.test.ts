import nixt from 'nixt'
import { join } from 'path'
const bin = join(__dirname, './../../../../', 'bin')
import { createClient, PlainClientAPI } from 'contentful-management'

const organizationId = process.env.CLI_E2E_ORG_ID
const accessToken = process.env.CONTENTFUL_INTEGRATION_TEST_CMA_TOKEN || ''

const app = () => {
  return nixt({ newlines: true }).cwd(bin).base('./contentful.js ').clone()
}

type Result = {
  stderr: string
  stdout: string
}

const cmd = 'organization export'

let cmaClient: PlainClientAPI
describe('organization import', () => {
  beforeAll(async () => {
    cmaClient = createClient({ accessToken }, { type: 'plain' })
  })
  afterAll(async () => {
    await Promise.allSettled([
      cmaClient.conceptScheme.delete({
        organizationId,
        conceptSchemeId: 'scheme0',
        version: 2
      }),
      cmaClient.concept.delete({
        organizationId,
        conceptId: 'concept2',
        version: 1
      }),
      cmaClient.concept.delete({
        organizationId,
        conceptId: 'concept1',
        version: 2
      }),
      cmaClient.concept.delete({
        organizationId,
        conceptId: 'concept0',
        version: 3
      })
    ])
  })

  test(`should create concepts and scheme if doesn't exist`, done => {
    app()
      .run(`${cmd} --organization-id ${organizationId}`)
      .expect(async ({ stdout }: Result) => {
        const resultText = stdout.trim()
        console.log(resultText)
      })
      .end(async () => {
        done()
      })
  })

  // test(`should update concept and scheme`, done => {
  //   app()
  //     .run(
  //       `${cmd} --organization-id ${organizationId} --content-file ../test/integration/cmds/organization/example-taxonomy-update.json`
  //     )
  //     .expect(async ({ stdout }: Result) => {
  //       const resultText = stdout.trim()

  //       expect(resultText).toContain('Create concepts')
  //       expect(resultText).toContain('Add concept relations')
  //       expect(resultText).toContain('Create concept schemes')
  //     })
  //     .end(async () => {
  //       const [concept0, scheme0] = await Promise.all([
  //         cmaClient.concept.get({
  //           conceptId: 'concept0',
  //           organizationId
  //         }),
  //         cmaClient.conceptScheme.get({
  //           conceptSchemeId: 'scheme0',
  //           organizationId
  //         })
  //       ])

  //       expect(concept0.prefLabel['en-US']).toContain('Updated Animals')
  //       expect(scheme0.prefLabel['en-US']).toContain('Updated Scheme')

  //       done()
  //     })
  // })
})
