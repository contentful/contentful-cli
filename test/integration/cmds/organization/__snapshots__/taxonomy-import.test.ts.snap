// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organization import should print help message: organization import help data is correct 1`] = `
"Usage: contentful organization import --content-file <path/to/content-file.json>

Options:
  -h, --help                Show help                                  [boolean]
  --management-token, --mt  Contentful management API token             [string]
  --organization-id, --oid  ID of Organization with source data
                                                             [string] [required]
  --header, -H              Pass an additional HTTP Header              [string]
  --content-file, -f        Content file with entities that need to be imported
                                                             [string] [required]
  --silent, -s              Suppress any log output   [boolean] [default: false]
  --error-log-file          Full path to the error log file             [string]"
`;
